From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 22 Dec 2020 18:01:36 +0100
Subject: wayland: Drop meta_wayland_compositor_repick()

This is now unused, thus the whole call chain from this function
to meta_wayland_pointer_repick() can be removed.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1654>
---
 src/wayland/meta-wayland-pointer.c | 22 ----------------------
 src/wayland/meta-wayland-pointer.h |  2 --
 src/wayland/meta-wayland-seat.c    |  9 ---------
 src/wayland/meta-wayland-seat.h    |  2 --
 src/wayland/meta-wayland.c         |  6 ------
 src/wayland/meta-wayland.h         |  2 --
 6 files changed, 43 deletions(-)

diff --git a/src/wayland/meta-wayland-pointer.c b/src/wayland/meta-wayland-pointer.c
index 71b3dbb..c2b8741 100644
--- a/src/wayland/meta-wayland-pointer.c
+++ b/src/wayland/meta-wayland-pointer.c
@@ -1008,28 +1008,6 @@ meta_wayland_pointer_start_popup_grab (MetaWaylandPointer      *pointer,
   return meta_wayland_popup_create (popup_surface, grab);
 }
 
-void
-meta_wayland_pointer_repick (MetaWaylandPointer *pointer)
-{
-  MetaBackend *backend = meta_get_backend ();
-  ClutterStage *stage = CLUTTER_STAGE (meta_backend_get_stage (backend));
-
-  graphene_point_t point;
-  ClutterActor *new_actor;
-
-  clutter_stage_get_device_coords (stage, pointer->device, NULL, &point);
-  new_actor =
-    clutter_stage_get_actor_at_pos (stage, CLUTTER_PICK_REACTIVE,
-                                    point.x, point.y);
-
-  clutter_stage_update_device (stage,
-                               pointer->device, NULL,
-                               point,
-                               CLUTTER_CURRENT_TIME,
-                               new_actor,
-                               TRUE);
-}
-
 void
 meta_wayland_pointer_get_relative_coordinates (MetaWaylandPointer *pointer,
 					       MetaWaylandSurface *surface,
diff --git a/src/wayland/meta-wayland-pointer.h b/src/wayland/meta-wayland-pointer.h
index 529177b..5eda527 100644
--- a/src/wayland/meta-wayland-pointer.h
+++ b/src/wayland/meta-wayland-pointer.h
@@ -124,8 +124,6 @@ MetaWaylandPopup *meta_wayland_pointer_start_popup_grab (MetaWaylandPointer
 
 void meta_wayland_pointer_end_popup_grab (MetaWaylandPointer *pointer);
 
-void meta_wayland_pointer_repick (MetaWaylandPointer *pointer);
-
 void meta_wayland_pointer_get_relative_coordinates (MetaWaylandPointer *pointer,
                                                     MetaWaylandSurface *surface,
                                                     wl_fixed_t         *x,
diff --git a/src/wayland/meta-wayland-seat.c b/src/wayland/meta-wayland-seat.c
index 258c0cb..f32173a 100644
--- a/src/wayland/meta-wayland-seat.c
+++ b/src/wayland/meta-wayland-seat.c
@@ -432,15 +432,6 @@ meta_wayland_seat_handle_event (MetaWaylandSeat *seat,
   return FALSE;
 }
 
-void
-meta_wayland_seat_repick (MetaWaylandSeat *seat)
-{
-  if (!meta_wayland_seat_has_pointer (seat))
-    return;
-
-  meta_wayland_pointer_repick (seat->pointer);
-}
-
 void
 meta_wayland_seat_set_input_focus (MetaWaylandSeat    *seat,
                                    MetaWaylandSurface *surface)
diff --git a/src/wayland/meta-wayland-seat.h b/src/wayland/meta-wayland-seat.h
index f41c442..ae4e107 100644
--- a/src/wayland/meta-wayland-seat.h
+++ b/src/wayland/meta-wayland-seat.h
@@ -69,8 +69,6 @@ gboolean meta_wayland_seat_handle_event (MetaWaylandSeat *seat,
 void meta_wayland_seat_set_input_focus (MetaWaylandSeat    *seat,
                                         MetaWaylandSurface *surface);
 
-void meta_wayland_seat_repick (MetaWaylandSeat *seat);
-
 gboolean meta_wayland_seat_get_grab_info (MetaWaylandSeat    *seat,
                                           MetaWaylandSurface *surface,
                                           uint32_t            serial,
diff --git a/src/wayland/meta-wayland.c b/src/wayland/meta-wayland.c
index 62f2195..4a7c883 100644
--- a/src/wayland/meta-wayland.c
+++ b/src/wayland/meta-wayland.c
@@ -131,12 +131,6 @@ meta_wayland_compositor_set_input_focus (MetaWaylandCompositor *compositor,
   meta_wayland_seat_set_input_focus (compositor->seat, surface);
 }
 
-void
-meta_wayland_compositor_repick (MetaWaylandCompositor *compositor)
-{
-  meta_wayland_seat_repick (compositor->seat);
-}
-
 static void
 wl_compositor_create_surface (struct wl_client   *client,
                               struct wl_resource *resource,
diff --git a/src/wayland/meta-wayland.h b/src/wayland/meta-wayland.h
index 3549fb2..e59bb97 100644
--- a/src/wayland/meta-wayland.h
+++ b/src/wayland/meta-wayland.h
@@ -55,8 +55,6 @@ void                    meta_wayland_compositor_update_key_state (MetaWaylandCom
                                                                   int                    key_vector_len,
                                                                   int                    offset);
 
-void                    meta_wayland_compositor_repick          (MetaWaylandCompositor *compositor);
-
 void                    meta_wayland_compositor_set_input_focus (MetaWaylandCompositor *compositor,
                                                                  MetaWindow            *window);
 
